test/e2e/acl_logging.go

```diff
@@ -0,0 +1,204 @@
+		By("creating pods")
+		cmd := []string{"/bin/bash", "-c", "/agnhost netexec --http-port 8000"}
+		for i := 0; i < 2; i++ {
+			pod := newAgnhostPod(fmt.Sprintf("pod%d", i+1), cmd...)
+			pod = fr.PodClient().CreateSync(pod)
+			Expect(waitForACLLoggingPod(fr, nsName, pod.GetName())).To(Succeed())
+			pods = append(pods, *pod)
+		}
+	})
+
+	BeforeEach(func() {
+		By("sending traffic between acl-logging test pods we trigger ACL logging")
+
+		clientPod := pods[pokerPodIndex]
+		pokedPod := pods[pokedPodIndex]
+		framework.Logf(
+			"Poke pod %s (on node %s) from pod %s (on node %s)",
+			pokedPod.GetName(),
+			pokedPod.Spec.NodeName,
+			clientPod.GetName(),
+			clientPod.Spec.NodeName)
+		Expect(
+			pokePod(fr, clientPod.GetName(), pokedPod.Status.PodIP)).To(HaveOccurred(),
+			"traffic should be blocked since we only use a deny all traffic policy")
+	})
+
+	AfterEach(func() {
+		pods = nil
+	})
+
+	It("the logs have the expected log level", func() {
+		clientPodScheduledPodName := pods[pokerPodIndex].Spec.NodeName
+		// Retry here in the case where OVN acls have not been programmed yet
+		Eventually(func() (bool, error) {
+			return assertLogs(
+				clientPodScheduledPodName,
+				nsName,
+				denyAllPolicyName,
+				initialDenyACLSeverity)
+		}, maxPokeRetries*pokeInterval, pokeInterval).Should(BeTrue())
+	})
+
+	When("the namespace's ACL logging annotation is updated", func() {
+		const updatedAllowACLLogSeverity = "debug"
+
+		BeforeEach(func() {
+			By(fmt.Sprintf("updating the namespace's ACL logging level to %s", updatedAllowACLLogSeverity))
+
+			namespace, err := fr.ClientSet.CoreV1().Namespaces().Get(context.Background(), nsName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "failed to retrieve the namespace")
+			Expect(setNamespaceACLLogSeverity(namespace, updatedAllowACLLogSeverity, updatedAllowACLLogSeverity)).To(Succeed())
+			namespace, err = fr.ClientSet.CoreV1().Namespaces().Get(context.Background(), nsName, metav1.GetOptions{})
+		})
+
+		BeforeEach(func() {
+			By("poking some more...")
+			clientPod := pods[pokerPodIndex]
+			pokedPod := pods[pokedPodIndex]
+
+			framework.Logf(
+				"Poke pod %s (on node %s) from pod %s (on node %s)",
+				pokedPod.GetName(),
+				pokedPod.Spec.NodeName,
+				clientPod.GetName(),
+				clientPod.Spec.NodeName)
+			Expect(
+				pokePod(fr, clientPod.GetName(), pokedPod.Status.PodIP)).To(HaveOccurred(),
+				"traffic should be blocked since we only use a deny all traffic policy")
+		})
+
+		It("the ACL logs are updated accordingly", func() {
+			clientPodScheduledPodName := pods[pokerPodIndex].Spec.NodeName
+			Eventually(func() (bool, error) {
+				return assertLogs(
+					clientPodScheduledPodName,
+					nsName,
+					denyAllPolicyName,
+					updatedAllowACLLogSeverity)
+			}, maxPokeRetries*pokeInterval, pokeInterval).Should(BeTrue())
+		})
+	})
+})
+
+func assertLogs(targetNodeName string, namespace string, policyName string, aclSeverity string) (bool, error) {
```

Comment:
I would put the function to collect ovn-controller logs in https://github.com/ovn-org/ovn-kubernetes/blob/master/test/e2e/util.go and then just verify the logs in-line